hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -319,6 +359,9 @@ public Bytes compute(final Bytes input, final MessageFrame messageFrame) {
 
 		if (isTokenReadOnlyTransaction) {
 			computeViewFunctionGasRequirement(messageFrame.getBlockValues().getTimestamp());
+		} else if (precompile instanceof TokenCreatePrecompile) {
```

Comment:
Refactoring of the whole `HTSPrecompiledContract` is planned as part of this [issue](https://github.com/hashgraph/hedera-services/issues/3079), which we aim to tackle in the next sprint, after HIP 358 and 376 are merged (to avoid dealing with conflicts). Perhaps we leave this as part of that issue?

Reply:
Yea, handle it in 3079